dogs/src/app/dogs-list/dogs-list.component.ts

```diff
@@ -18,6 +19,12 @@ export class DogsListComponent implements OnInit {
 	}
 
 	private prepareDogList(): void {
-		this.dogNames = this.apiService.getDogs()
+		this.apiService.getDogNames().subscribe(data => {
+			this.responseData = data;
+			const dogNamseList = this.responseData.message;
+			for (let dogName in dogNamseList) {
+				this.dogNames.push(dogName);
+			}
```

Comment:
```suggestion
			for (let dogName of data.message) {
				this.dogNames.push(dogName);
			}
```
does it work with for-in?

Reply:
Apparently "for key in object" is the way to iterate through the properties of an object (each breed is a property of "message" object according to this specific API)